pysmt/test/test_typechecker.py

```diff
@@ -192,5 +192,16 @@ def test_examples(self):
         for (f, _, _, _) in get_example_formulae():
             self.assertIs(f.get_type(), BOOL)
 
+    def test_suspend(self):
+        with self.assertRaises(PysmtTypeError):
+            And(self.x, self.r)
+
+        r = None
+        with SuspendTypeChecking():
+            r = And(self.x, self.r)
+
```

Comment:
Adding this test here would fail:
```python
        with self.assertRaises(PysmtTypeError):
            And(self.x, self.r)
```
Although, I think we agree that it should not.